### DISCUSSION TITLE
Experimental Offline Standalone Mac Installer/Application for Stable Diffusion Web UI (unofficial) 20230919

### DISCUSSION THREAD

**Discussion by brkirch** [brkirch/stable-diffusion-webui#42] [Announcements]
This is an offline standalone installer/application for a modified version of AUTOMATIC1111's Stable Diffusion web UI that includes experimental changes and uses a release candidate PyTorch build for 2.1.0. Note that this is an test build so crashes and bugs should be expected. All dependencies needed to start using Stable Diffusion (other than the model) are included and this does not require Xcode command line tools to be installed.

This release currently only has the installer and application for Apple silicon Macs, an installer and application for Intel Macs will be included on a later release. macOS 13.5 or later is highly recommended, older versions of macOS are untested.

This has the following changes and bug fixes since [20230915_experimental](https://github.com/brkirch/stable-diffusion-webui/releases/tag/20230915_experimental):
- The installation of additional extension dependencies is now supported
- Restarting Web UI now works correctly when using the `--electron` option or application
- Updating extensions from the UI in the application is fixed

<hr /><em>This discussion was created from the release <a href='https://github.com/brkirch/stable-diffusion-webui/releases/tag/20230919_experimental'>Experimental Offline Standalone Mac Installer/Application for Stable Diffusion Web UI (unofficial) 20230919</a>.</em>

**Comment by byewind1** [comment]
Can a prompt button be added before closing? Accidental actions often lead to the program being shut down.

**Comment by tiancool** [comment]
How not to automatically generate Logs is turned off by default. And can I minimize the terminal window when starting the software? Another unusable plug-in: sd-dynamic-prompts,sdweb-easy-prompt-selector, thank you for your efforts

**Comment by sohotrightnow** [comment]
These Standalone Mac installers have been fantastic for me using an M1 macbook.  One question: why don't my extensions seem to work?  I can see them load in Terminal, they're all listed in Extensions and checked, but they never appear in the interface.  Eg. ADetailer, Roop, ReActor.  All appear in the Automatic 1111 interface when I do the install through Homebrew but not with these standalone installers.

**Reply by Infiexe** [reply]
ReActor works for me no problem, ADetailer in the other hand... Doesn't install its requirement, 'rich' module and I although I've installed it through homebrew, this standalone experimental webUI version doesn't see it. I guess it uses local python and rich must be installed somewhere locally too, but IDK how to do it on MacOS and where to install it

**Comment by jayododds** [comment]
It looks like this release has trouble with the lobe theme as well as showing all embeddings in the UI for some reason. It only recognizes 4 out of about 30+ in my folders and it randomly chooses which ones to surface from a few different subfolders. It really didn't like it when I tried to use the lobe theme to refresh the folder structure of any of the Lora/models/embeddings - threw up all sorts of errors about the dictionary changing size mid operation etc. Uninstalling the theme helped but the embeddings still aren't showing up in the embeddings tab.

**Comment by Max1mu1** [comment]
Hi! Has anyone encountered this? IP adapter Controlnet send errors
Tryed this links but alas:
https://github.com/Mikubill/sd-webui-controlnet/issues/2131
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/13818


2023-11-13 20:48:38,525 - ControlNet - INFO - Loading model: ip-adapter_sd15 [dbbc7cfe]
2023-11-13 20:48:38,526 - ControlNet - INFO - Loaded state_dict from [/Users/macmini/stable-diffusion-webui/models/ControlNet/ip-adapter_sd15.safetensors]
*** Error running process: /Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py
    Traceback (most recent call last):
      File "/Users/macmini/stable-diffusion-webui/modules/scripts.py", line 623, in process
        script.process(p, *script_args)
      File "/Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 977, in process
        self.controlnet_hack(p)
      File "/Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 966, in controlnet_hack
        self.controlnet_main_entry(p)
      File "/Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 688, in controlnet_main_entry
        model_net = Script.load_control_model(p, unet, unit.model)
      File "/Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 321, in load_control_model
        model_net = Script.build_control_model(p, unet, model)
      File "/Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 350, in build_control_model
        network = build_model_by_guess(state_dict, unet, model_path)
      File "/Users/macmini/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet_model_guess.py", line 242, in build_model_by_guess
        raise '[ControlNet Error] Cannot recognize the ControlModel!'
    TypeError: exceptions must derive from BaseException

---
100%|███████████████████████████████████████████| 20/20 [00:10<00:00,  1.83it/s]
Total progress: 100%|███████████████████████████| 20/20 [00:10<00:00,  1.95it/s]
Total progress: 100%|███████████████████████████| 20/20 [00:10<00:00,  1.95it/s]

**Comment by SnowyField1906** [comment]
```
launch.py: error: unrecognized arguments: --electron
```
Sorry sir, this happens when I put `--electron` in `.sh`. Am I wrong somewhere?

**Comment by haythamodeh** [comment]
Im getting this error when loading adetailer:
Error loading script: !adetailer.py
Traceback (most recent call last):
      File "/Users/-----/Desktop/Stable Diffusion Web UI.app/Contents/Resources/stable-diffusion-webui/modules/scripts.py", line 382, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "/Users/-----/Desktop/Stable Diffusion Web UI.app/Contents/Resources/stable-diffusion-webui/modules/script_loading.py", line 10, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "/Users/-----/Documents/Stable Diffusion Web UI/extensions/adetailer/scripts/!adetailer.py", line 17, in <module>
        from rich import print
    ModuleNotFoundError: No module named 'rich'

I tried installing rich with pip, still didn't work, I also tried updating requirements.txt and requirements_versions.txt.
i tried deleting venv folder, but got stuck with some issues so I restored the venv back, I'm not sure why this problem is happening. any help would be appreciated. thanks

**Reply by Infiexe** [reply]
Same here! Did you resolve it?

**Reply by nhhyung** [reply]
Open 'webui-macos-env.sh' with TextEdit and remove the '-skip-install' option at the 'COMMANDLINE_ARGS' line

**Comment by PippoApps** [comment]
Is it possible to start the app with --api settings on?

**Comment by ZeroLu** [comment]
Hi everyone, how does this achieve higher performance on M1/M2? Does it use CoreML(https://machinelearning.apple.com/research/stable-diffusion-coreml-apple-silicon)?
Highly appreciated if anyone can help! I am really tired of generating one image with more than 6 minutes.... :-(

**Comment by iliavls** [comment]
Hello! Sorry for bothering you, but will it be possible to make an SD app the same as yours with electron and special version of Torch, also using the Forge backend (https://github.com/lllyasviel/stable-diffusion-webui-forge.git)? And will it be more efficient than your version?

**Reply by curiosus0ne** [reply]
That would be great, forge web seems to be so much better in terms of performance!

**Comment by bijlevel** [comment]
I added model 'v2-1_768-ema-pruned.ckpt', but the model does not load because of error 'TypeError: 'NoneType' object is not callable'. What am I doing wrong? I'm using v 20230919.

**Reply by bijlevel** [reply]
I think I know now the answer myself. The Web UI is based on SD v1 and the model in question needs (obviously?) v2.1...
